[MME] Add no_time_zone_in_emm_information param

osmith42 opened the pull request:
Allow network operators to omit the local time zone in the EMM information. This is useful for better compatibility with some UEs.

According to 3GPP TS 124.301 Table 8.2.13.1, the parameter is optional.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/open5gs/open5gs?pullRequest=2745) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/open5gs/open5gs?pullRequest=2745) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/open5gs/open5gs?pullRequest=2745) it.</sub>

acetcom commented:
@osmith42 

Why were lines 323-349 of emm-handler.c modified?
And also, that parameter is not applied to 5G Core.

Please let me know if you have any other questions.

Thanks a lot!
Sukchan

osmith42 commented:
Hi @acetcom, thanks for the quick review!

> Why were lines 323-349 of emm-handler.c modified?

The patch moves all code related to filling out the local time zone IE (and daylight saving time IE) between the `if {…}`, so it only runs if the new option is not enabled.

> And also, that parameter is not applied to 5G Core.

I've updated the patch to do this for 5G as well, and renamed the parameter to `no_time_zone_information` as it is no longer EMM information specific.

Could you take another look at it?

acetcom commented:
@osmith42 

Thank you for making the modifications up to the 5G Core part as well.
Sukchan
